doc/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+# Provide a local project and cmake_minimum_required calls
+# to permit to generate the documentation without configuring
+# the complete project
+
+cmake_minimum_required(VERSION 3.0)
+
+# This variable should contain the argument of the project call in the main CMakeLists.txt
+set(ORIGINAL_PROJECT_NAME "LibTemplateCMake")
+project(${ORIGINAL_PROJECT_NAME}Docs)
+
+include(GNUInstallDirs)
+
+find_package(Doxygen QUIET)
+
+if(DOXYGEN_FOUND)
+  # This variables are used inside the Doxyfile.in template
```

Comment:
To be compliant with the other CMake files, it would be better to have 4 spaces indentation instead of 2.
I know that CMake suggests 2 spaces, but it's really ugly to me.
If we want to adhere to the 2 spaced convention, we may accept this file, but edit all the others accordingly.

Reply:
I see, let's go for the consistency. I was automatically using 2-spaces due to https://github.com/robotology/yarp/blob/master/.github/CONTRIBUTING.md#cmake .

Reply:
Eventually it would be super to have also a cmake_format (https://github.com/cheshirekow/cmake_format) configuration style, but this is another story.

Reply:
> Eventually it would be super to have also a cmake_format (https://github.com/cheshirekow/cmake_format) configuration style, but this is another story.

This is THE story. Does CMake have an internal prittifier? We may use a custom_command for it.